src/multibody/geometry.hxx

```diff
@@ -42,10 +43,9 @@ namespace pinocchio
   GeomIndex GeometryModel::addGeometryObject(const GeometryObject & object,
                                              const ModelTpl<S2,O2,JointCollectionTpl> & model)
   {
-    assert( //TODO: reenable when relevant (object.parentFrame == -1) ||
-           (model.frames[object.parentFrame].type == pinocchio::BODY)  );
-    assert( //TODO: reenable when relevant (object.parentFrame == -1) ||
-           (model.frames[object.parentFrame].parent == object.parentJoint) );
+    if(object.parentFrame < (FrameIndex)model.nframes)
+      PINOCCHIO_CHECK_INPUT_ARGUMENT(model.frames[object.parentFrame].parent == object.parentJoint,
+                                     "The object joint parent and its frame joint parent do not match.");
```

Comment:
Shouldn't you throw something if `object.parentFrame >= (FrameIndex)model.nframes` ?

Reply:
the `object.parentFrame` is optional. And as it is a `FrameIndex == size_t`, the default value cannot be `-1`.
I decided to set it to the max value.
So, we should not check this condition.